packages/frame/src/react/index.js

```diff
@@ -0,0 +1,211 @@
+
+    if (sidenavOverlayed) return variants.overlay
+
+    if (sidenavOpen) {
+      return mediumMedia ? variants.open : variants.overlay
+    } else {
+      return mediumMedia ? variants.minimized : variants.closed
+    }
+  }, [sidenav, sidenavOpen, sidenavOverlayed, mediumMedia])
+
+  const [variant, setVariant] = useState(getVariant)
+  useEffect(() => {
+    setVariant(getVariant)
+  }, [getVariant])
+
+  const skipTargetId = 'ps-frame-skip-target'
+  const skipTargetRef = useRef()
+
+  const focusSkipTarget = useCallback(() => {
+    if (!skipTargetRef.current) return
+
+    skipTargetRef.current.focus()
+  }, [])
+
+  return (
+    <div
+      ref={ref}
+      {...styles.frame(themeName, props)}
+      {...filterReactProps(props)}
+    >
+      <Theme name={themes.dark}>
+        <SkipBanner href={'#' + skipTargetId} />
+      </Theme>
+
+      <Theme name={themes.dark}>
+        <div {...styles.topnav()}>{topnav}</div>
+      </Theme>
+
+      <Container variant={variant}>
+        {sidenav && <SideNav variant={variant}>{sidenav}</SideNav>}
+
+        <main {...styles.content()}>
+          <SkipTarget
+            id={skipTargetId}
+            onClick={focusSkipTarget}
+            ref={skipTargetRef}
+          />
+
+          {children}
+        </main>
+      </Container>
+    </div>
+  )
+})
+
+Frame.displayName = 'Frame'
+Frame.defaultProps = {
+  sidenavOpen: false,
+  sidenavOverlayed: false
+}
+
+Frame.propTypes = {
+  children: PropTypes.oneOfType([PropTypes.func, PropTypes.node]),
+  sidenav: PropTypes.oneOfType([PropTypes.func, PropTypes.node]),
+  sidenavOpen: PropTypes.bool,
+  sidenavOverlayed: PropTypes.bool,
+  topnav: PropTypes.oneOfType([PropTypes.func, PropTypes.node])
+}
+
+function SkipBanner(props) {
+  return (
+    <div {...styles.skipBanner()} {...props}>
+      <Button
+        appearance={Button.appearances.secondary}
+        size={Button.sizes.small}
+        tabIndex={0}
+      >
+        skip to main content
+      </Button>
+    </div>
+  )
+}
+SkipBanner.propTypes = {
+  href: PropTypes.string.isRequired
+}
+
+const SkipTarget = React.forwardRef((props, ref) => {
+  return <a ref={ref} tabIndex={-1} {...props} />
+})
+SkipTarget.displayName = 'SkipTarget'
+SkipTarget.propTypes = {
+  id: PropTypes.string.isRequired
+}
+
+function Container(props) {
+  const { variant, ...rest } = props
+
+  return <div {...styles.container(variant)} {...rest} />
+}
+Container.propTypes = {
+  children: PropTypes.node,
+  variant: PropTypes.oneOf(Object.keys(vars.sidenavVariants))
+}
+
+function SideNav(props) {
+  const { children, variant, ...rest } = props
+
+  const ref = React.useRef()
+
+  const lockScroll = variant === vars.sidenavVariants.overlay
+  useBodyScrollLock(ref, lockScroll)
+
+  return (
+    <Theme name={themes.dark}>
+      <div ref={ref} {...styles.sidenav(variant)} {...rest}>
+        <div {...styles.sidenavOverflowMask()}>
+          <Scrollable {...styles.sidenavInner()}>{children}</Scrollable>
```

Comment:
Will overflow-y and Scrollable interfere with each other?  2 scroll bars, maybe?
https://github.com/pluralsight/design-system/blob/master/packages/verticaltabs/src/css/index.js#L45

Scrolling in this example, at least, looks ok:

![scrollingOk](https://user-images.githubusercontent.com/552044/89328595-d9e7ea00-d64a-11ea-9b69-e18aa89eb01d.gif)

Reply:
it should be fine but i'll double check